Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -448,10 +448,42 @@ - (void)_uninstallScreenTimeWebpageController
     if (!_screenTimeWebpageController)
         return;
 
+    if (_screenTimeBlurredSnapshot) {
+        [_screenTimeBlurredSnapshot removeFromSuperview];
+        _screenTimeBlurredSnapshot = nil;
+    }
+
     [[_screenTimeWebpageController view] removeFromSuperview];
     [_screenTimeWebpageController removeObserver:self forKeyPath:@"URLIsBlocked" context:&screenTimeWebpageControllerBlockedKVOContext];
     _screenTimeWebpageController = nil;
 }
+
+- (void)_showScreenTimeShieldForWindow
+{
+    bool showShield = [self window] ? true : false;
+
+    if (showShield) {
+        if (![_configuration _showsSystemScreenTimeBlockingView] && _screenTimeBlurredSnapshot)
+            [_screenTimeBlurredSnapshot setHidden:NO];
+        else if ([_configuration _showsSystemScreenTimeBlockingView])
+            [[_screenTimeWebpageController view] setHidden:NO];
+    } else {
+        if (_screenTimeBlurredSnapshot)
+            [_screenTimeBlurredSnapshot setHidden:YES];
+        else if ([_configuration _showsSystemScreenTimeBlockingView])
+            [[_screenTimeWebpageController view] setHidden:YES];
+    }
+
+    // To not track URLs while offscreen, we need to set suppress usage recording if either in private browsing or not in window (which includes being occluded on macOS).
+#if PLATFORM(MAC)
+    BOOL isNowVisible = [self window] && (([self window].occlusionState & NSWindowOcclusionStateVisible) == NSWindowOcclusionStateVisible);
```

Comment:
Nit: Why using `BOOL` for this?

Reply:
I think we should prefer `BOOL` in pure Obj-C code...